File: backend/src/main/java/com/dream/services/FactoryService.java
package com.dream.services;

import org.springframework.data.domain.Pageable;

import org.springframework.data.domain.Page;

import com.dream.models.Factory;

public interface FactoryService {
	
	Page<Factory> getFactories(Pageable pageable);
	
	Factory getFactoryById(long id);
	
	Factory createFactory(Factory Factory);
	
	Factory updateFactory(long id, Factory Factory);
	
	void deleteFactory(long id);
}

File: frontend/src/app/services/administration/factory.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Factory } from '../../model/factory.model';
import { API_URL } from '../../app.constants';

@Injectable({
  providedIn: 'root'
})
export class FactoryService {

  constructor(
    private http:HttpClient
  ) { }

  retrieveAllFactories(page,size,sort) {
    return this.http.get<Factory[]>(`${API_URL}/factories?page=${page}&size=${size}&sort=${sort}`);
  }

  deleteFactory(factoryId){
    return this.http.delete(`${API_URL}/factories/${factoryId}`);
  }

  retrieveFactory(factoryId){
    return this.http.get<Factory>(`${API_URL}/factories/${factoryId}`);
  }

  updateFactory(factoryId, factory){
    return this.http.put(
          `${API_URL}/factories/${factoryId}`
                , factory);
  }

  createFactory(factory){
    return this.http.post(
              `${API_URL}/factories`
                , factory);
  }
}
